src/app/core/user/entity-user/entity-user.component.ts

```diff
@@ -0,0 +1,176 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  computed,
+  inject,
+  Input,
+  OnInit,
+  signal,
+} from "@angular/core";
+import { DynamicComponent } from "../../config/dynamic-components/dynamic-component.decorator";
+import { AlertService } from "../../alerts/alert.service";
+import { HttpClient } from "@angular/common/http";
+import { UntilDestroy } from "@ngneat/until-destroy";
+import { SessionSubject } from "../../session/auth/session-info";
+import { Entity } from "../../entity/model/entity";
+import { switchMap } from "rxjs/operators";
+import { environment } from "../../../../environments/environment";
+import { UserAdminService } from "../user-admin-service/user-admin.service";
+import { UserAccount } from "../user-admin-service/user-account";
+import { of } from "rxjs";
+import {
+  UserDetailsComponent,
+  UserDetailsAction,
+} from "../user-details/user-details.component";
+
+/**
+ * Display User Account details and configuration related to a given profile Entity.
+ */
+@UntilDestroy()
+@DynamicComponent("EntityUser")
+@DynamicComponent("UserSecurity") // for backwards compatibility. Prefer to use "EntityUser"
+@Component({
+  selector: "app-entity-user",
+  templateUrl: "./entity-user.component.html",
+  styleUrls: ["./entity-user.component.scss"],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  imports: [UserDetailsComponent],
+})
+export class EntityUserComponent implements OnInit {
+  private readonly userAdminService = inject(UserAdminService);
+  private readonly alertService = inject(AlertService);
+  private readonly http = inject(HttpClient);
+  private readonly sessionInfo = inject(SessionSubject);
+
+  @Input() entity?: Entity;
+  user = signal<UserAccount | null>(null);
+  editing = computed(
+    () => this.formMode() === "edit" || this.formMode() === "create",
+  );
+  userIsPermitted = signal<boolean>(false);
+  formMode = signal<"create" | "edit" | "view">("create");
```

Comment:
I don't see for what we are using this formMode here at all (it seems user-details handles all form-related things internally? which is good).
Am I missing something, @Abhinegi2 ?